### PR TITLE
Do dynamic rollout for the pull workflow

### PR DESCRIPTION
**Pull request by DanilBaibak**
Do dynamic rollout for the pull workflow.

### CODE REVIEW

**Review by jeanschmidt — APPROVED** [review]

File: .github/workflows/_android-build-test.yml [hunk 1]
```diff
@@ -23,14 +23,19 @@ on:
...
+        description: Runner type
 
 env:
   GIT_DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
 
 jobs:
   filter:
     if: github.repository_owner == 'pytorch'
-    runs-on: [self-hosted, linux.large]
+    runs-on: ${{ inputs.runner }}
```
Comment (ZainRizvi) [inline .github/workflows/_android-build-test.yml:38]: why drop the `self-hosted` label?

File: .github/workflows/_win-build.yml [hunk 2]
```diff
@@ -43,7 +48,7 @@ jobs:
   build:
     # Don't run on forked repos.
     if: github.repository_owner == 'pytorch'
-    runs-on: [self-hosted, windows.4xlarge.nonephemeral]
+    runs-on: ${{ inputs.runner }}
```
Comment (ZainRizvi) [inline .github/workflows/_win-build.yml:51]: windows runners don't work on LF accounts yet. We need to get custom Windows AMIs working there first

File: .github/workflows/pull.yml [hunk 3]
```diff
@@ -345,79 +380,93 @@ jobs:
...
       cuda-version: cpu
       sync-tag: win-cpu-build
       test-matrix: |
         { include: [
-          { config: "default", shard: 1, num_shards: 3, runner: "windows.4xlarge.nonephemeral" },
-          { config: "default", shard: 2, num_shards: 3, runner: "windows.4xlarge.nonephemeral" },
-          { config: "default", shard: 3, num_shards: 3, runner: "windows.4xlarge.nonephemeral" },
+          { config: "default", shard: 1, num_shards: 3, runner: "${{ needs.get-label-type.outputs.label-type }}windows.4xlarge.nonephemeral" },
+          { config: "default", shard: 2, num_shards: 3, runner: "${{ needs.get-label-type.outputs.label-type }}windows.4xlarge.nonephemeral" },
+          { config: "default", shard: 3, num_shards: 3, runner: "${{ needs.get-label-type.outputs.label-type }}windows.4xlarge.nonephemeral" },
```
Comment (ZainRizvi) [inline .github/workflows/pull.yml:393]: windows doesn't work yet

File: .github/workflows/trunk.yml [hunk 4]
```diff
@@ -170,15 +177,17 @@ jobs:
...
       cuda-version: cpu
       sync-tag: win-cpu-build
       test-matrix: |
         { include: [
-          { config: "default", shard: 1, num_shards: 3, runner: "windows.4xlarge.nonephemeral" },
-          { config: "default", shard: 2, num_shards: 3, runner: "windows.4xlarge.nonephemeral" },
-          { config: "default", shard: 3, num_shards: 3, runner: "windows.4xlarge.nonephemeral" },
+          { config: "default", shard: 1, num_shards: 3, runner: "${{ needs.get-label-type.outputs.label-type }}windows.4xlarge.nonephemeral" },
+          { config: "default", shard: 2, num_shards: 3, runner: "${{ needs.get-label-type.outputs.label-type }}windows.4xlarge.nonephemeral" },
+          { config: "default", shard: 3, num_shards: 3, runner: "${{ needs.get-label-type.outputs.label-type }}windows.4xlarge.nonephemeral" },
```
Comment (ZainRizvi) [inline .github/workflows/trunk.yml:190]: windows doesn't work yet

**Review by ZainRizvi — CHANGES_REQUESTED** [review]
Before merging, let's make sure the `slow.yml` jobs run successfully with the new LF runners and remove any references to windows runners which, are not expected to work yet.

There's also something weird going on with the windows runners. I re-enabled the LF runners issue and retried the windows build job, bit it still appears to have it's old label, and thus gets directed towards the Meta runners instead of the LF ones

File: .github/workflows/_android-build-test.yml (hunk 1 above)
Comment (DanilBaibak) [reply .github/workflows/_android-build-test.yml:38]: Two reasons:
* The `[self-hosted, ${{ inputs.runner }}]` syntax does not work.
* We use special labels - this means that runners are self-hosted. Therefore, we can skip the `self-hosted` label.

**Comment by DanilBaibak** [comment]
@ZainRizvi, The general idea is to merge this PR and then work stabilization of LF infra. Means you can remove yourself [from the list](https://github.com/pytorch/test-infra/issues/5132) and work on other pytorch parts if you need.

**Pull request closed without merging**